EDU-2283: Fix typo in "Production bundling" documentation, add another diagnostic scenario (#2838)
temporalio/documentation: docs/dev-guide/tscript/debugging.mdx

#### Production bundling

You may experience your Client sending stripped names as the Workflow "Type" when scheduling a Workflow.
If you are getting errors like this:

```
Error: 3 INVALID_ARGUMENT: WorkflowType is not set on request.
```

> Or, if you are seeing stripped-down Workflow types in the Temporal Service's Web UI like this:

Author: ```suggestion
Or, if you are seeing stripped-down Workflow "Type" listings in the Temporal Service's Web UI like this:
```

> Or, if you are seeing stripped-down Workflow types in the Temporal Service's Web UI like this:

Reviewer: I'm not sure there's enough context here - the alt text of the image actually does explain what we're looking for, but I'd take some of that prose and explain what's specifically different here. 

Webpack changed the Workflow's function name to something shorter, and now Temporal doesn't know how to handle it.

Author: Updated as suggested

> ![Temporal Web UI showing stripped "Workflow Type" entries, in this case the single letter 's'](/img/stripped_workflow_types_in_webui.png)

Author: ```suggestion
![Temporal Web UI showing stripped Workflow "Type" entries, in this case the single letter 's'](/img/stripped_workflow_types_in_webui.png)
```

It can be due to your bundler stripping out Workflow function names.

> Temporal relies on those names to set the "Workflow Type" in the Service Web UI.

Author: ```suggestion
Temporal relies on those names to set the Workflow "Type" in the Temporal Service Web UI.
```

> Disable that by configuring webpack (in webpack.config.js) and it should work.

Reviewer: ```suggestion
To prevent the build process from removing Workflow function names,  modify the webpack configuration file ( `webpack.config.js`) by setting `keep_fnames` to `true` in the `TerserPlugin` configuration section:
```

#### Production bundling

You may experience your Client sending stripped names as the Workflow "Type" when scheduling a Workflow.
Webpack can change the Workflow's function name to something shorter.
Temporal won't know how to handle the mismatch between the shorter name and the expect Workflow type.

You may experience errors like this:

```
Error: 3 INVALID_ARGUMENT: WorkflowType is not set on request.
```

Or you may see shorter names in the Temporal Service's Web UI when Webpack changed the Workflow's function name to something shorter, in this case the single letter 's':

![Temporal Web UI showing stripped "Workflow Type" entries, in this case the single letter 's'](/img/stripped_workflow_types_in_webui.png)

This issue can happen when your bundler strips out Workflow function names.
Temporal relies on those names to set the "Workflow Type" in the Service Web UI.

> To prevent the build process from removing Workflow function names,  modify the webpack configuration file ( `webpack.config.js`) to set the Boolean that retains the original names in the `TerserPlugin` configuration section.

Reviewer: ```suggestion
To prevent the build process from shortening Workflow function names,  modify the webpack configuration file ( `webpack.config.js`) to set the Boolean that retains the original names in the `TerserPlugin` configuration section.
```

Changed to be more accurate, as it does not "remove" them. Oops.

> Setting this option (`keep_fnames`) to `true` prevents name stripping.

Reviewer: ```suggestion
Setting the option (`keep_fnames`) to `true` prevents name stripping.
```

});
```

See [esbuild docs](https://esbuild.github.io/api/#keep-names) for more information: https://esbuild.github.io/api/#keep-names
</TabItem>
</Tabs>

> Another option to prevent this bundle issue is to overwrite the function name with `Object.defineProperty`:

Reviewer 2: I would remove this part, see @mjameswh 's comment in the Jira ticket

Author: Done. Thank you.

});
```

> See [esbuild docs](https://esbuild.github.io/api/#keep-names) for more information: https://esbuild.github.io/api/#keep-names

Reviewer: ```suggestion
See [esbuild docs](https://esbuild.github.io/api/#keep-names) for more information.
```

});
```

> See [esbuild docs](https://esbuild.github.io/api/#keep-names) for more information.

Reviewer: ```suggestion
See the [esbuild docs](https://esbuild.github.io/api/#keep-names) for more information.
```
